src/HttpReply.cc

```diff
@@ -115,6 +115,14 @@ HttpReply::pack() const
     return mb;
 }
 
+HttpReply *
+HttpReply::makeConnectionEstablished() {
+
+    auto rep = new HttpReply;
```

Comment:
Just in case this code gets into official long-term code, please make this code exception-safe. It should not leak rep if set() throws. The method should still return raw pointer. Using unique_ptr may be a good option here.